api/opds.py

```diff
@@ -241,17 +241,20 @@ def lane_url(self, lane):
         # sublanes. Otherwise it will take the user to a list of the
         # books in the lane by author.
 
-        if not lane.sublanes and not lane.parent:
+        if lane and isinstance(lane, Lane) and lane.sublanes:
+            url = self.groups_url(lane)
+        elif lane and (
+            isinstance(lane, Lane)
+            or isinstance(lane, WorkBasedLane)
+            or isinstance(lane, ContributorLane)
+            or isinstance(lane, SeriesLane)
+            ):
```

Comment:
This is a big list that we might forget to add to in the future. Can you create a superclass of all these custom worklists, or invert the logic and create a special class for the top-level WorkList?

Reply:
Any suggestions on a name for the superclass?

Reply:
Maybe DynamicLane? They form OPDS lanes, but they're not ORM objects like Lane.